projects/storefrontlib/src/cms-components/checkout/components/schedule-replenishment-order/schedule-replenishment-order.component.ts

```diff
@@ -0,0 +1,213 @@
+  ScheduleReplenishmentForm,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { ICON_TYPE } from '../../../misc/icon/icon.model';
+import { CheckoutReplenishmentFormService } from '../../services/checkout-replenishment-form-service';
+
+@Component({
+  selector: 'cx-schedule-replenishment-order',
+  templateUrl: './schedule-replenishment-order.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ScheduleReplenishmentOrderComponent implements OnInit, OnDestroy {
+  private subscription: Subscription = new Subscription();
+
+  iconTypes = ICON_TYPE;
+  orderTypes = ORDER_TYPE;
+  daysOfWeek = Object.keys(DaysOfWeek).map((key) => DaysOfWeek[key]);
+  recurrencePeriodType = Object.keys(recurrencePeriod).map(
+    (key) => recurrencePeriod[key]
+  );
+
+  selectedOrderType$: Observable<
+    ORDER_TYPE
+  > = this.checkoutService.getCurrentOrderType();
+
+  isMonthly: Boolean = false;
+  isWeekly: Boolean = false;
+  currentDaysOfWeek: DaysOfWeek[] = [];
+  numberOfDays: string[];
+  numberOfWeeks: string[];
+  currentDate: string;
+  scheduleReplenishmentFormData: ScheduleReplenishmentForm;
+
+  constructor(
+    protected checkoutService: CheckoutService,
+    protected checkoutReplenishmentFormService: CheckoutReplenishmentFormService
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.checkoutReplenishmentFormService
+        .getScheduleReplenishmentFormData()
+        .subscribe((data) => {
+          this.scheduleReplenishmentFormData = data;
+        })
+    );
+
+    this.initConfig();
+  }
+
+  changeOrderType(orderType: ORDER_TYPE): void {
+    this.checkoutService.setOrderType(orderType);
+  }
+
+  changeNumberOfDays(nDays: string): void {
+    this.checkoutReplenishmentFormService.setScheduleReplenishmentFormData({
+      ...this.scheduleReplenishmentFormData,
+      numberOfDays: nDays,
+    });
+  }
+
+  changeNumberOfWeeks(nWeeks: string): void {
+    this.checkoutReplenishmentFormService.setScheduleReplenishmentFormData({
+      ...this.scheduleReplenishmentFormData,
+      numberOfWeeks: nWeeks,
+    });
+  }
+
+  changeRecurrencePeriodType(type: string): void {
+    this.isWeekly = type === recurrencePeriod.WEEKLY;
+    this.isMonthly = type === recurrencePeriod.MONTHLY;
+
+    this.checkoutReplenishmentFormService.setScheduleReplenishmentFormData({
+      ...this.scheduleReplenishmentFormData,
+      recurrencePeriod: type,
+    });
+  }
+
+  changeDayOfTheMonth(dayOfMonth: string): void {
+    this.checkoutReplenishmentFormService.setScheduleReplenishmentFormData({
+      ...this.scheduleReplenishmentFormData,
+      nthDayOfMonth: dayOfMonth,
+    });
+  }
+
+  changeReplenishmentStartDate(date: string): void {
+    if (Boolean(date)) {
+      this.numberOfDays = this.createNumberStringArray(
+        this.getNumberOfDayInAMonth(date)
+      );
+
+      if (
+        Number(this.scheduleReplenishmentFormData.nthDayOfMonth) >
```

Comment:
:)

Reply:
Won't this be handle with the my-company date picker?

Reply:
nope don't think so as it doesn't associate with the number of days input and day of the month input

Reply:
using defaults as in accelerator as Bill mentioned